sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -232,11 +232,14 @@ def read_all_items(
         if populate_query_metrics is not None:
             feed_options["populateQueryMetrics"] = populate_query_metrics
 
+        if hasattr(response_hook, 'clear'):
+            response_hook.clear()
```

Comment:
I don't love this. An alternative would be to provide an ABC base class for subclassing if  someone wants a hook that can clear state here.